tests/unit/manage/test_views.py

```diff
@@ -1657,3 +1657,26 @@ def test_get(self, db_request):
             "project": project,
             "journals": [newer_journal, older_journal],
         }
+
+    def test_raises_400_with_pagenum_type_str(self, monkeypatch, db_request):
+        params = MultiDict({"page": "abc"})
+        db_request.params = params
+
+        journals_query = pretend.stub()
+        db_request.journals_query = pretend.stub(
+            journals_query=lambda *a, **kw: journals_query
+        )
+
+        page_obj = pretend.stub(page_count=10, item_count=1000)
+        page_cls = pretend.call_recorder(lambda *a, **kw: page_obj)
+        monkeypatch.setattr(views, "SQLAlchemyORMPage", page_cls)
+
+        url_maker = pretend.stub()
+        url_maker_factory = pretend.call_recorder(lambda request: url_maker)
+        monkeypatch.setattr(views, "paginate_url_factory", url_maker_factory)
+
+        project = ProjectFactory.create()
+        with pytest.raises(HTTPBadRequest):
+            views.manage_project_history(project, db_request)
+
+        assert page_cls.calls == []
```

Comment:
Would you mind adding two additional positive tests?

Ideally creating >25 JournalEntry objects and asserting the first 25 entries are returned when retrieving the first page and asserting that the remaining entry is returned when requesting the second page.

Or less entries if you decide to move the 25 to a setting

Reply:
I added the two tests you suggested plus a test for the case where someone requests a page out of range.  For example, requesting page 3 when there are only 2 pages.